### PR TITLE
Migrate to develocity plugin

### PR DESCRIPTION
**Pull request by Goooler**
https://docs.gradle.com/enterprise/gradle-plugin/legacy/#develocity_migration

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
-val isCiBuild = System.getenv("CI") != null
-
-gradleEnterprise {
+develocity {
     buildScan {
-        termsOfServiceUrl = "https://gradle.com/terms-of-service"
-        termsOfServiceAgree = "yes"
-        publishAlwaysIf(isCiBuild)
+        termsOfUseUrl = "https://gradle.com/terms-of-service"
```
Comment (TWiStErRob) [inline settings.gradle.kts:8]: hmm, did the url change?

https://docs.gradle.com/enterprise/gradle-plugin/current/#connecting_to_scans_gradle_com

or is this something else?

File: settings.gradle.kts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+        termsOfUseAgree = "yes"
+        // TODO: workaround for https://github.com/gradle/gradle/issues/22879.
+        val isCI = providers.environmentVariable("CI").isPresent
+        publishing.onlyIf { isCI }
```
Comment (TWiStErRob) [inline settings.gradle.kts:12]: They have exactly the same example here:

https://docs.gradle.com/enterprise/gradle-plugin/current/#publishing_based_on_criteria

Could we stick with `getenv` as they show? I agree providers are the way, but if not working, then I would stick with the old way.

is it possible there may be a different syntax, something like:

```
publishing.onlyIf(providers.environmentVariable("CI").orElse(false))
```

File: settings.gradle.kts (hunk 1 above)
Comment (Goooler) [reply settings.gradle.kts:8]: `termsOfServiceUrl` has been replaced by `termsOfUseUrl`.

File: settings.gradle.kts (hunk 2 above)
Comment (Goooler) [reply settings.gradle.kts:12]: It can't be inline, this is why I commented TODO here.

File: settings.gradle.kts (hunk 1 above)
Comment (TWiStErRob) [reply settings.gradle.kts:8]: I meant the value of it, see the docs.

Comment (TWiStErRob) [reply settings.gradle.kts:8]: It looks like both are accepted for now.

File: settings.gradle.kts (hunk 2 above)
Comment (Goooler) [reply settings.gradle.kts:12]: Inlining `!System.getenv("CI").isNullOrEmpty()` is working but `providers.environmentVariable("CI").isPresent`, I can use the former if you prefer.

File: settings.gradle.kts (hunk 1 above)
Comment (TWiStErRob) [reply settings.gradle.kts:8]: Let's use the new one:
```suggestion
        termsOfUseUrl = "https://gradle.com/help/legal-terms-of-use"
```
It's in the develocity plugin docs + if you make a typo in it, it'll recommend this one.

File: settings.gradle.kts [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
-
-gradleEnterprise {
+develocity {
     buildScan {
-        termsOfServiceUrl = "https://gradle.com/terms-of-service"
-        termsOfServiceAgree = "yes"
-        publishAlwaysIf(isCiBuild)
+        termsOfUseUrl = "https://gradle.com/help/legal-terms-of-use"
+        termsOfUseAgree = "yes"
+        publishing.onlyIf { System.getenv("CI") != null }
```
Comment (TWiStErRob) [inline settings.gradle.kts:10]:
```suggestion
        uploadInBackground = false
        publishing.onlyIf { System.getenv("CI") != null }
```

I think we need this based on:
https://docs.gradle.com/enterprise/gradle-plugin/current/#configuring_background_uploading

And I think this is not related to the migration, it should have been on before too (same docs in legacy too). @szpak did you ever have a problem where a scan wasn't available?

**Review by TWiStErRob — APPROVED** [review]
@Goooler Looks good, thank you for looking and doing it!

@szpak please have a look at the scope creep, or merge and we can discuss a separate PR.

File: settings.gradle.kts (hunk 3 above)
Comment (szpak) [reply settings.gradle.kts:10]: I don't recall that situation, however, I don't often use build scans, so I might miss it.

Comment (szpak) [reply settings.gradle.kts:10]: Maybe because there is some time consumed do save local cache. Probably we should set it to `false`, as you suggested.

**Review by szpak — APPROVED** [review]
